app/models/finance/billing_strategy.rb

```diff
@@ -166,7 +166,7 @@ def create_invoice(opts = {})
   end
 
   def create_invoice_counter(period)
-    invoice_prefix = (billing_monthly? ? period : period.begin.year).to_param
+    invoice_prefix = period.to_param[0..(billing_monthly? ? 6 : 3)]
```

Comment:
can you explain where the 6 and 3 come from?

Reply:
Sure, it's either the 7 (0..6) or the 4 (0..3) initial characters of the string, respectively to monthly billing friendly ids (YYYY-MM-########) and yearly billing friendly ids (YYYY-########).

Reply:
ok, maybe leave inline comment as this is not very expressive and represents business logic.

Reply:
Done.